java/kotlin-extractor2/src/main/kotlin/entities/Class.kt

```diff
@@ -56,7 +62,7 @@ fun KotlinFileExtractor.extractClassSource(
                 }
             }
 
-            val locId = tw.getLocation(c.psiSafe() ?: TODO())
+            val locId = c.psiSafe<KtElement>()?.let { tw.getLocation(it) } ?: tw.getFileOnlyLocation()
```

Comment:
I don't think we want to do this. If we can see the source, then we'll get the location when we extract the source. If we're here, we should just give the whole-file location.

Reply:
I think this is that place, though? In KE1 at least this function was shared between the extract-kt-source-class and extract-class-file paths, and it is shared here too since it works at symbol level. I'm not aware of another place where the whole class would get a source-file location ascribed if we know of one.

Reply:
Ah, I was thinking of the context of the PR, and didn't realise which function we were in. I think it'll be cleaner to refactor this slightly differently: If we're starting from the PSI source, then I don't think we want to go to the symbol and then back to the PSI again. But we can think about that in another PR.